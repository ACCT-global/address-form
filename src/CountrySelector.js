import React, { Component } from 'react'
import PropTypes from 'prop-types'
import CountryType from './propTypes/CountryType'

class CountrySelector extends Component {
  render() {
    const { country, shipsTo, onChangeSelectedCountry } = this.props

    return (
      <div className="country-selector">
        <label>
          Pa√≠s
          <select value={country} onChange={onChangeSelectedCountry}>
            {shipsTo.map(country => (
              <option value={country} key={country}>
                {country}
              </option>
            ))}
          </select>
        </label>
      </div>
    )
  }
}

CountrySelector.propTypes = {
  country: CountryType,
  shipsTo: PropTypes.array.isRequired,
  onChangeSelectedCountry: PropTypes.func.isRequired,
}

export default CountrySelector
