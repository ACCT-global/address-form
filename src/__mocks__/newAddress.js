export default {
  addressId: '1',
  addressType: 'residential',
  city: null,
  complement: null,
  country: null,
  geoCoordinates: null,
  neighborhood: null,
  number: null,
  postalCode: null,
  receiverName: null,
  reference: null,
  state: null,
  street: null,
}
