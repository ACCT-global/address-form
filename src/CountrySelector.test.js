import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import CountrySelector from './CountrySelector'
import renderer from 'react-test-renderer'
import newAddress from './__mocks__/newAddress'

describe('CountrySelector', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <CountrySelector
        address={newAddress}
        shipsTo={['BRA', 'USA']}
        onChangeAddress={jest.fn()}
      />,
      div
    )
  })

  it('show options', () => {
    const tree = renderer
      .create(
        <CountrySelector
          address={newAddress}
          shipsTo={['BRA', 'USA']}
          onChangeAddress={jest.fn()}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('default value', () => {
    const tree = renderer
      .create(
        <CountrySelector
          address={{
            ...newAddress,
            country: 'BRA',
          }}
          shipsTo={['BRA', 'USA']}
          onChangeAddress={jest.fn()}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('react to change value', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <CountrySelector
        address={{
          ...newAddress,
          country: 'BRA',
        }}
        shipsTo={['BRA', 'USA']}
        onChangeAddress={handleChange}
      />
    )

    const event = { target: { value: 'USA' } }
    wrapper.find('select').simulate('change', event)

    expect(handleChange).toHaveBeenCalledWith({
      ...newAddress,
      country: 'USA',
    })
  })

  it('shold clean postalCode when country changes', () => {
    const handleChange = jest.fn()
    const wrapper = shallow(
      <CountrySelector
        address={{
          ...newAddress,
          country: 'BRA',
          postalCode: '123',
        }}
        shipsTo={['BRA', 'USA']}
        onChangeAddress={handleChange}
      />
    )

    const event = { target: { value: 'USA' } }
    wrapper.find('select').simulate('change', event)

    expect(handleChange).toHaveBeenCalledWith({
      ...newAddress,
      country: 'USA',
      postalCode: null,
    })
  })
})
