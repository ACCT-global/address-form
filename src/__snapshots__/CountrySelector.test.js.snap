// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default value 1`] = `
<div
  className="country-selector"
>
  <select
    onChange={[Function]}
    value="BRA"
  >
    <option
      value="BRA"
    >
      BRA
    </option>
    <option
      value="USA"
    >
      USA
    </option>
  </select>
</div>
`;

exports[`show options 1`] = `
<div
  className="country-selector"
>
  <select
    onChange={[Function]}
    value={undefined}
  >
    <option
      value="BRA"
    >
      BRA
    </option>
    <option
      value="USA"
    >
      USA
    </option>
  </select>
</div>
`;
